### ISSUE TITLE
ERC Token Decimals Support

### ISSUE THREAD

**Issue by branciard** [poanetwork/bridge-ui#159]
Problem : 
Today only Token with 18 decimals are supported because amount are formatted like this : 
```
Web3Utils.toWei(amount)
```
Solution proposal : 
Before formating we can retrieve the token decimals in config or in contract : 
```
const decimals = await contract.methods.decimals().call()
```

To format here a list of potential solutions  : 

1. use existing web3 lib and create a revert map in bridge-ui like this : 

```
const web3UtilsRevertunitMap = {
    '1':         'wei',
    '3':         'kwei',
    '6':         'mwei',
    '9':         'nano',
    '12':        'micro',
    '15':        'milli',
    '18':        'ether',
    '24':        'mether',
    '27':        'gether',
    '30':        'tether'
};

```
and code somthing like this : 
```
 var decimalsToWeb3unit = web3UtilsRevertunitMap[decimals];
  return Web3Utils.fromWei(totalSupply,decimalsToWeb3unit)
```
This solution will only support token decimals :    1, 3,6,9,12,15,18,24,27,30.

2.  use another lib ether.js (not good for ui size optimization ?) : 
 https://docs.ethers.io/ethers.js/html/api-utils.html?highlight=utils#ether-strings-and-wei
```
utils . parseUnits ( valueString , decimalsOrUnitName )   =>   BigNumber
utils . formatUnits ( wei , decimalsOrUnitName )   =>   string
```
This solution will only support token decimals :    1, 3,6,9,12,15,18,24,27,30.

3. implement a new specific fonction in web3js that support custom decimal token formating.

4. implement a new specific fonction in bridge-ui that support custom decimal token formating.

5. other solution ?

Which solution do you think is the best to implement ?

I can then develop it and submit a PR for this token decimals support.

**Comment by branciard** [comment]
Ok to implement a specific function to format all kind of decimals. 

I suppose that this decimals format must be done only in erc-erc mode. In Native-erc or erc-native, it must be and still remains 18 decimals erc only, right?

**Comment by akolotov** [comment]
I think it make sense to support any number of decimals in every bridge mode. It makes sense also use a default value of decimals as 18 if the contract does not provide the corresponding interface by some reasons.

**Comment by branciard** [comment]
Thank you for the feedback. Ok for a global token decimal support with 18 decimals default as backup. I will work on it and let you informed.

**Comment by akolotov** [comment]
Addressed in #165
